### ISSUE TITLE
Functions like DATEADD, DATEDIFF, etc. work incorrectly when datepart is passed as variable

### ISSUE THREAD

**Issue by glebkozh** [linq2db/linq2db#1615]
Hello!
I've got the following issue: when I try to use Sql.DateAdd, Sql.DateDiff, etc. with datepart passed as variable instead of literal expression, I've got underlying DateParts enum int values used as variable values in generated SQL as these functions do not have InlineParameters switched on.

Here is simplifed example.

Sample entity:
```c#
        [Table]
        private class Entity
        {
            [Identity, PrimaryKey]
            public int Id { get; set; }
            [Column]
            public DateTime Created { get; set; }
            [Column]
            public DateTime Deleted { get; set; }
        }
```
Two examples: 
```c#
var query1 = _db.GetTable<Entity>().Where(e => e.Deleted > Sql.DateAdd(Sql.DateParts.Year, 1, e.Created));
var datePart = Sql.DateParts.Year;
var query2 = _db.GetTable<Entity>().Where(e => e.Deleted > Sql.DateAdd(datePart, 1, e.Created));
```
Generated SQL: 
```sql
BeforeExecute
--  SqlServer SqlServer.2012
SELECT
	[e].[Id],
	[e].[Created],
	[e].[Deleted]
FROM
	[Entity] [e]
WHERE
	[e].[Deleted] > DateAdd(Year, 1, [e].[Created])

BeforeExecute
--  SqlServer SqlServer.2012
DECLARE @datePart Int -- Int32
SET     @datePart = 0

SELECT
	[e].[Id],
	[e].[Created],
	[e].[Deleted]
FROM
	[Entity] [e]
WHERE
	[e].[Deleted] > DateAdd(@datePart, 1, [e].[Created])
```
The second case fails because of invalid first parameter of DateAdd function.

Thanks!
### Environment details
linq2db version: _2.6.4_
Database Server: _Sql Server 13.0.1601.5_
Database Provider: _Sql.Data.SqlClient 4.6.27110.4_
Operating system: _Windows 10 Pro 10.0.17134_
Framework version: _.NET Core 2.2.102_

**Comment by sdanyliv** [comment]
Working on this in https://github.com/linq2db/linq2db/pull/780
Will be fixed soon.

**Comment by MaceWindu** [comment]
Fixed by #780
